digits/model/tasks/caffe_train.py

```diff
@@ -1156,14 +1200,25 @@ def can_view_weights(self):
         return False
 
     @override
-    def infer_one(self, data, snapshot_epoch=None, layers=None, gpu=None):
+    def infer_one(self,
```

Comment:
It's a tough one :-) Like you, I think the first one looks better but the fact that the second one is more robust to diffs is appealing too. If I was forced to cast a vote I think I'd go for the first one (I'm tempted to think that beauty is more important than robust diffs).